optuna/samplers/tpe/sampler.py

```diff
@@ -448,6 +448,20 @@ def _log_normal_cdf(cls, x, mu, sigma):
 
 def _get_observation_pairs(study, param_name):
     # type: (InTrialStudy, str) -> List[Tuple[float, Tuple[float, float]]]
+    """Get observation pairs from the study.
+
+       This function collects observation pairs from the trials of the study.
+       The trials that doesn't contain the parameter named ``param_name`` are excluded
```

Comment:
Thank you for adding the comments. I understood the behavior and the usage of `_get_observation_pairs` clearly.

I found a small mistake here.
`The trials that doesn't contain` seems to be a typo for `The trials that don't contain`.

Reply:
I already fixed the mistake https://github.com/pfnet/optuna/pull/439/commits/8f697a5b8579f99a5987273a4a341eafe56f9dc9.
Sorry for my carelessness.